### PR TITLE
remove claim that data_files does not work with wheels

### PR DESCRIPTION
**Pull request by graingert-coef**
## Summary of changes

remove claim that data_files does not work with wheels, it does work with wheels, the data goes into the `distribution-1.0.data/data` dir which then gets copied into the installation prefix. What doesn't work is: "Distutils allows directory to be an absolute installation path, but this is discouraged since it is incompatible with the wheel packaging format"

### Pull Request Checklist
- [ ] Changes have tests
- [ ] News fragment added in [`newsfragments/`].
  _(See [documentation][PR docs] for details)_

[`newsfragments/`]: https://github.com/pypa/setuptools/tree/master/newsfragments
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: docs/references/keywords.rst [hunk 1]
```diff
@@ -183,8 +183,7 @@ extensions).
 
 ``data_files``
     .. warning::
-        ``data_files`` is deprecated. It does not work with wheels, so it
-        should be avoided.
+        ``data_files`` is deprecated.
```
Comment (abravalheri) [inline docs/references/keywords.rst:186]: Should we then say something like the following:

```
This is an advanced feature and it is *not intended to work with absolute installation paths*.
All files listed in ``data_files`` will be installed in paths relative to a directory
decided by the package installer (e.g. `pip`),
which usually results in nesting under a virtual environment.
See :docs:`userguide/datafiles` for an alternative placing inside the package directory.
Please do not use this setting for things like man-pages, application launchers or anything that requires system-wide installation.
```

Comment (graingert-coef) [reply docs/references/keywords.rst]: I updated the warning with this

File: docs/references/keywords.rst [hunk 2]
```diff
@@ -183,8 +183,12 @@ extensions).
 
 ``data_files``
     .. warning::
-        ``data_files`` is deprecated. It does not work with wheels, so it
-        should be avoided.
+        This is an advanced feature and it is *not intended to work with absolute installation paths*.
+        All files listed in ``data_files`` will be installed in paths relative to a directory
+        decided by the package installer (e.g. `pip`),
+        which usually results in nesting under a virtual environment.
+        See :docs:`userguide/datafiles` for an alternative placing inside the package directory.
```
Comment (graingert-coef) [inline docs/references/keywords.rst:190]: this is raising /home/runner/work/setuptools/setuptools/docs/references/keywords.rst:186: ERROR: Unknown interpreted text role "docs".

https://github.com/pypa/setuptools/actions/runs/7820343433/job/21334826952?pr=4210#step:5:92

Comment (graingert-coef) [reply docs/references/keywords.rst:190]:
```suggestion
        See :doc:`userguide/datafiles` for an alternative placing inside the package directory.
```

File: docs/references/keywords.rst [hunk 3]
```diff
@@ -183,8 +183,12 @@ extensions).
 
 ``data_files``
     .. warning::
-        ``data_files`` is deprecated. It does not work with wheels, so it
-        should be avoided.
+        This is an advanced feature and it is *not intended to work with absolute installation paths*.
+        All files listed in ``data_files`` will be installed in paths relative to a directory
+        decided by the package installer (e.g. `pip`),
+        which usually results in nesting under a virtual environment.
+        See :doc:`userguide/datafiles` for an alternative placing inside the package directory.
```
Comment (graingert-coef) [inline docs/references/keywords.rst:190]: now this is raising /home/runner/work/setuptools/setuptools/docs/references/keywords.rst:186: WARNING: unknown document: 'userguide/datafiles'

Comment (graingert-coef) [reply docs/references/keywords.rst:190]:
```suggestion
        See :doc:`/userguide/datafiles` for an alternative placing inside the package directory.
```

Comment (abravalheri) [reply docs/references/keywords.rst]: Thank you for spotting that and for the changes!

**Comment by abravalheri** [comment]
Hi @graingert-coef, I expanded a little more on the note.
Please let me know if you disagree, otherwise I think we can merge this.

File: docs/references/keywords.rst [hunk 4]
```diff
@@ -182,13 +182,32 @@ extensions).
...
+    .. attention::
+        **DISCOURAGED** - This is an advanced feature and it is
+        *not intended to work with absolute paths*.
+        All files listed in ``data_files`` will be installed in paths relative
+        to a directory decided by the package installer (e.g. `pip`).
+        This usually results in nesting under a virtual environment.
+        We **STRONGLY ADVISE AGAINST** using this setting for things like
+        application launchers, desktop files or anything that requires
+        system-wide installation [#manpages]_, unless you have extensive
+        experience in Python packaging and has carefully considered all the
```
Comment (abravalheri) [inline docs/references/keywords.rst:194]:
```suggestion
        experience in Python packaging and have carefully considered all the
```

**Comment by abravalheri** [comment]
Thank you very much @graingert-coef

**Pull request merged**